src/report_workflow/test_run_runner.py

```diff
@@ -0,0 +1,136 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import urllib.request
+from typing import Any
+from urllib.error import HTTPError
+
+import validators
+import yaml
+
+from manifests.test_manifest import TestManifest
+from manifests.test_run_manifest import TestRunManifest
+from report_workflow.report_args import ReportArgs
+
+
+class TestRunRunner:
+    args: ReportArgs
+    test_manifest: TestManifest
+    tests_dir: str
+    test_run_manifest: TestRunManifest
+    test_run_data: dict
+
+    def __init__(self, args: ReportArgs, test_manifest: TestManifest) -> None:
+        self.args = args
+        self.base_path = args.base_path
+        self.test_manifest = test_manifest
+        self.test_run_data = self.test_run_manifest_data_template("manifest")
+        self.product_name = test_manifest.__to_dict__().get("name")
+        self.name = self.product_name.replace(" ", "-").lower()
+        self.components = self.args.components
+        self.test_run_id = args.test_run_id
+        self.test_type = self.args.test_type
+        self.test_manifest_path = self.args.test_manifest_path
+        self.artifact_paths = ""
+        for k, v in self.args.artifact_paths.items():
+            self.artifact_paths = " ".join([self.artifact_paths, k + "=" + v]).strip(" ")
+
+        self.dist_manifest = "/".join([self.args.artifact_paths[self.name], "dist", self.name, "manifest.yml"])
```

Comment:
Is this path or url? If its path, you might want to join using path else this will fail on windows.

Reply:
Good catch! Ideally in CI level, this is the URL. While it makes sense that if we are running this workflow after local test workflow, the `artifacts_path` and all test results would be local path.

Reply:
@gaiksaya Just updated the distribution manifest path here with specific test cases.